widget/net.lua

```diff
@@ -66,9 +68,56 @@ local function factory(args)
             dev_now.last_t   = now_t
             dev_now.last_r   = now_r
 
+	    if wifi_info == "on" and os.execute(string.format("ls /sys/class/net/%s/wireless", dev)) and string.match(dev_now.carrier, "1") then
+            dev_now.wifi = true
+
+            helpers.line_callback(string.format("iw dev %s link", dev), function(line)
```

Comment:
This process is being respawned at each `net.update()` timeout. Since it is asynchronous, it will likely cause a memory leak.